libpod/healthcheck.go

```diff
@@ -32,11 +32,12 @@ func (r *Runtime) HealthCheck(name string) (define.HealthCheckStatus, error) {
 	}
 
 	hcStatus, err := checkHealthCheckCanBeRun(container)
-	if err == nil {
-		hcStatus, err := container.runHealthCheck()
-		if err := container.processHealthCheckStatus(hcStatus); err != nil {
-			return hcStatus, err
-		}
+	if err != nil {
+		return hcStatus, err
+	}
+
+	hcStatus, err = container.runHealthCheck()
```

Comment:
That's too big of a shotgun and would cause dead locks with on-failure actions.  We also don't know how long a health check may run, so we should be careful to not lock too long.  AFAICS the race is on reading and writing the health-check logs, so we should lock there.

I prefer deferring the locking fix to another PR where we can shave the yak.

Reply:
I was probably too hasty.  The other caller is locked.  I'll fix it.

Reply:
Used `HealthCheckStatus` to lock when reading the log file.

Reply:
That fixes the reading part in your code but `runHealthCheck()` still calls `updateHealthCheckLog()` which does not look for writing and reading.

Also looking from a performance point it would make much more sense to return the full log in updateHealthCheckLog() instead of reading it again

Reply:
Thanks for you persistence!  The value is passed back and the races are fixed.